arm-cognitiveservices/2017-04-18/swagger/cognitiveservices.json

```diff
@@ -0,0 +1,1137 @@
+          "$ref": "#/definitions/OperationDisplayInfo",
+          "description": "The operation supported by Cognitive Services."
+        },
+        "origin": {
+          "description": "The origin of the operation.",
+          "type": "string"
+        },
+        "properties": {
+          "description": "Additional properties.",
+          "type": "object"
+        }
+      }
+    },
+    "OperationDisplayInfo": {
+      "description": "The operation supported by Cognitive Services.",
+      "type": "object",
+      "properties": {
+        "description": {
+          "description": "The description of the operation.",
+          "type": "string"
+        },
+        "operation": {
+          "description": "The action that users can perform, based on their permission level.",
+          "type": "string"
+        },
+        "provider": {
+          "description": "Service provider: Microsoft Cognitive Services.",
+          "type": "string"
+        },
+        "resource": {
+          "description": "Resource on which the operation is performed.",
+          "type": "string"
+        }
+      }
+    },
+    "CheckSkuAvailabilityParameter": {
+      "description": "Check SKU availability parameter.",
+      "properties": {
+        "skus": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Sku"
+          },
+          "description": "The SKU of the resource."
+        },
+        "kind": {
+          "$ref": "#/definitions/CognitiveServicesAccountKind",
+          "description": "The Kind of the resource."
+        },
+        "type": {
+          "type": "string",
+          "description": "The Type of the resource."
+        }
+      }
+    },
+    "CheckSkuAvailabilityResultList": {
+      "description": "Check SKU availability result list.",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CheckSkuAvailabilityResult"
+          },
+          "description": "Check SKU availability result list."
+        }
+      }
+    },
+    "CheckSkuAvailabilityResult": {
+      "description": "Check SKU availability result.",
+      "properties": {
+        "kind": {
+          "$ref": "#/definitions/CognitiveServicesAccountKind",
+          "description": "The Kind of the resource."
+        },
+        "type": {
+          "type": "string",
+          "description": "The Type of the resource."
+        },
+        "skuName": {
+          "$ref": "#/definitions/Sku",
+          "description": "The SKU of Cognitive Services account."
+        },
+        "skuAvailable": {
+          "type": "boolean",
+          "description": "Indicates the given SKU is available or not."
+        },
+        "reason": {
+          "type": "string",
+          "description": "Reason why the SKU is not available."
+        },
+        "message": {
+          "type": "string",
+          "description": "Additional error message."
+        }
```

Comment:
@veronicagg , I noticed that validate-example reports warnings for these 2 fields because the in example they are null. I tried to make the type ["string", "null"] but then AutoRest reports errors then.
Please advice how to make tools happy in this case.